### DISCUSSION TITLE
Converting Spacy Sentence Embeddings Back to Text

### DISCUSSION THREAD

**Discussion by mbdzi** [explosion/spaCy#11782] [Help: Coding & Implementations] [answered]
My questions:

1. Is it possible for one to convert Spacy document vectors back to text? If so, how may it be done? I looked through the Spacy documentation and could not locate any hints of what I would need to do to effect the vector-to-string reverse transformation.
2. Supposing that I conducted mathematical operations on Spacy document vectors, would it be possible to convert the resultant vector to the closest string approximation?

My code thus far:

```
import spacy
nlp = spacy.load("en_core_web_lg")

string_1 = "I love to hike during weekends"
string_2 = "I love to play Frisbee with my dog during weekends"

doc_1 =  nlp(string_1)
doc_2 =  nlp(string_2)

doc_1_vector = doc_1.vector
doc_2_vector = doc_2.vector

weight_1 = 0.6
weight_2 = 0.4

doc_1_doc_2_average_vector = np.average(np.vstack((doc_1_vector, doc_2_vector)), axis=0, weights=[weight_1, weight_2])

# Is it possible to convert "doc_1_doc_2_average_vector" to the closest string approximation? If so, how can I do this?
```

Otherwise, thank you Ines and team for creating a great product!

**Comment by polm** [comment, marked as answer]
No, the transformation to vectors is not reversible, or designed to be so. 

For pipelines like `en_core_web_lg` the document vector is an average of token vectors, so it's the same even if you re-order the text, for example. 

If you have saved a bunch of documents and their vectors, you can query for the most similar vector and then get the string attached to that.
